janus.c

```diff
@@ -191,16 +191,42 @@ static gchar *local_ip = NULL;
 gchar *janus_get_local_ip(void) {
 	return local_ip;
 }
-static gchar *public_ip = NULL;
+static GPtrArray *public_ip = NULL;
+guint janus_get_public_ip_count(void) {
+	return public_ip ? public_ip->len : 0;
+}
+gchar *janus_get_public_ip_index(guint index) {
+	if (!public_ip || !public_ip->len) {
+		/* Fallback to the local IP, if we have no public one */
+		return local_ip;
+	}
+	if (index >= public_ip->len) {
+		return NULL;
+	}
+	return g_ptr_array_index(public_ip, index);
+}
 gchar *janus_get_public_ip(void) {
-	/* Fallback to the local IP, if we have no public one */
-	return public_ip ? public_ip : local_ip;
+	return janus_get_public_ip_index(0);
+}
+void janus_set_public_ip_list(GPtrArray *ips) {
+	/* once set do not override */
+	if (public_ip != NULL) {
+		return;
+	}
+	public_ip = g_ptr_array_ref(ips);
 }
-void janus_set_public_ip(const char *ip) {
+static void janus_free_string(gpointer data) {
+	g_free(data);
+}
```

Comment:
This is useless: if this is to have something to pass to `g_ptr_array_new_with_free_func`, you can just pass `g_free`, as we do for other glib destroy functions.